pkgs/servers/foundationdb/cmake.nix

```diff
@@ -31,23 +32,28 @@ let
           inherit rev sha256;
         };
 
-        buildInputs = [ libressl boost ];
-        nativeBuildInputs = [ cmake ninja python3 openjdk mono ]
+        buildInputs = [ ssl boost ]
+          ++ lib.optionals (lib.versionAtLeast version "7.1.0") [ msgpack toml11 ];
+
+        nativeBuildInputs = [ pkg-config cmake ninja python3 openjdk mono ]
           ++ lib.optionals useClang [ llvmPackages.lld ];
 
         separateDebugInfo = true;
         dontFixCmake = true;
 
         cmakeFlags =
-          [ "-DCMAKE_BUILD_TYPE=Release"
-            (lib.optionalString officialRelease "-DFDB_RELEASE=TRUE")
+          [ (lib.optionalString officialRelease "-DFDB_RELEASE=TRUE")
 
-            # FIXME: why can't libressl be found automatically?
-            "-DLIBRESSL_USE_STATIC_LIBS=FALSE"
-            "-DLIBRESSL_INCLUDE_DIR=${libressl.dev}"
-            "-DLIBRESSL_CRYPTO_LIBRARY=${libressl.out}/lib/libcrypto.so"
-            "-DLIBRESSL_SSL_LIBRARY=${libressl.out}/lib/libssl.so"
-            "-DLIBRESSL_TLS_LIBRARY=${libressl.out}/lib/libtls.so"
+            # Disable CMake warnings for project developers.
+            "-Wno-dev"
+
+            # CMake Error at fdbserver/CMakeLists.txt:332 (find_library):
+            # >   Could not find lz4_STATIC_LIBRARIES using the following names: liblz4.a
+            "-DSSD_ROCKSDB_EXPERIMENTAL=FALSE"
+
+            # FoundationDB's CMake is hardcoded to pull in jemalloc as an external
```

Comment:
Likewise, we can patch the build system to look for a system-provided jemalloc, but I don't know enough CMake to make that happen.